Userland/DevTools/HackStudio/Debugger/Debugger.cpp

```diff
@@ -190,11 +195,11 @@ int Debugger::debugger_loop()
             do_step_over(regs);
             return Debug::DebugSession::DebugDecision::Continue;
         case DebuggerAction::Exit:
-            // NOTE: Is detaching from the debuggee the best thing to do here?
-            // We could display a dialog in the UI, remind the user that there is
-            // a live debugged process, and ask whether they want to terminate/detach.
             dbgln("Debugger exiting");
-            return Debug::DebugSession::DebugDecision::Detach;
+            // In case the debug session is exited while execution is already paused.
+            m_on_continue_callback();
```

Comment:
I may be missing something - why do we call `m_on_continue_callback` here?

Reply:
If the debug session is stopped while execution is paused, the hack studios debugger state doesn't properly clean up (namely, current execution point arrow). I could have either copied that code from the continue callback to the stop block or ran the continue callback, which IMO feels more correct.

Reply:
Oh okay, thanks for clarifying that for me.
Personally I would prefer to copy the relevant code that resets the UI into the exit callback (its only 3 lines so still feels reasonable).
Calling the continue callback here seems kinda confusing and could also cause future trouble if we change the logic of the continue callback to include something we would not want to do when the debugger exits.

Reply:
Okay, sounds reasonable :)